lib/winapi/findfile.c

```diff
@@ -0,0 +1,131 @@
+#include <string.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <winbase.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <hal/fileio.h>
+#include <hal/winerror.h>
+#include <fileapi.h>
+
+struct FileInfo {
+    FILE_DIRECTORY_INFORMATION dirInfo;
+    // Reserve path buffer minus the null-terminator and the first byte provided by dirInfo
```

Comment:
Huh?! Why is it even `[1]` in dirinfo? is `[]` as trailing member an extension?

Reply:
Nevermind, this is how MSDN defines it.. - but then I wonder how / why it should be used in the real world. Our current solution looks very hacky.

Reply:
Are you sure that this doesn't have to include the zero terminator though? As it has a dedicated `dirInfo->FileNameLength` field the source string doesn't have to be zero-terminated. So this field probably needs room for the additional zero termination.